src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -272,7 +272,7 @@ private void AddOriginToResult(string origin, CorsPolicy policy, CorsResult resu
             {
                 result.AllowedOrigin = origin;
 
-                if(policy.Origins.Count > 1)
+                if(policy.Origins.Count > 1 || policy.Origins.Any(o => o.Contains("*")))
```

Comment:
IMO, this isn't correct. The behavior for `IsOriginAllowed` is allowed to be user defined which means the user code could have just as likely said "IsOriginAllowed = origin => origin.EndsWith("blah.com");` which would make this check incorrect. We should add a property to `CorsPolicy` that allows configuring this and use the line above as the default implementation.

i.e. `CorsPolicy.IsVaryByOriginRequired => origin => Origins.Count > 1 || origin.Contains("*")`